.github/sync_labels.py

```diff
@@ -0,0 +1,348 @@
+        the removement in case it represents the stat of a PR.  In all other
+        cases add the successor of the label except if there is none or there
+        exists another label of the corresponding list.
+        """
+        sel_list = selection_list(label)
+        if not sel_list:
+            return
+
+        if label in self.get_labels():
+            # this is possible if two labels of the same selection list
+            # have been removed in one step (via multiple selection in the
+            # pull down menue). In this case `label` has been added
+            # on the `on_label_remove` of the first of the two labels.
+            partn = self.active_partners(label)
+            if not partn:
+                self.add_comment('Label *%s* can not be removed (last one of list)!' % label)
+            else:
+                self.on_label_add(partn[0])
+            return
+
+        item = sel_list(label)
+        if sel_list is State and self._pr:
+            if item in [State.positive_review, State.needs_work]:
+                self.add_comment('Label *%s* can not be removed. Please use the corresponding functionality of GitHub' % label)
+                self.select_label(label)
+                return
+
+        if not self.active_partners(label):
+            succ = sel_list(label).succ()
+            if succ:
+                # add the next weaker label
+                self.select_label(succ.value)
+            else:
+                self.add_comment('Label *%s* can not be removed since there is no replacement' % label)
+                self.select_label(label)
+            
+
+###############################################################################
+# Main
+###############################################################################
+cmdline_args = sys.argv[1:]
+
+getLogger().setLevel(INFO)
+info('cmdline_args (%s) %s' % (len(cmdline_args), cmdline_args))
+
+if len(cmdline_args) < 4:
+    print('Need 4 arguments: action, url, label, rev_state' )
+    exit
+else:
+    action, url, label, rev_state = cmdline_args
+
+info('action: %s' % action)
+info('url: %s' % url)
+info('label: %s' % label)
+info('rev_state: %s' % rev_state)
+
+gh = GhLabelSynchronizer(url)
+
+if action == 'opened':
+    gh.add_default_label(Priority.major.value)
+    gh.add_default_label(IssueType.enhancement.value)
+    if gh.is_pull_request():
+        if not gh.is_draft():
+            gh.add_default_label(State.needs_review.value)
+
+if action == 'reopened':
+    if gh.is_pull_request():
+        if not gh.is_draft():
+            gh.add_default_label(State.needs_review.value)
+
+if action == 'closed':
+    for lab in State:
+        gh.remove_label(lab.value)
+
+if action == 'labeled':
+    gh.on_label_add(label)
+
+if action == 'unlabeled':
+    gh.on_label_remove(label)
+
+if action == 'submitted':
+    if rev_state == 'approved':
+        gh.select_label(State.positive_review.value)
```

Comment:
Can we check here if all existing reviews are approving, and only then add "positive review"? Otherwise it gets confusing if you have one "changes requested" and one "approving" review.

Reply:
> Can we check here if all existing reviews are approving, and only then add "positive review"? Otherwise it gets confusing if you have one "changes requested" and one "approving" review.

Of course we should! Good point!

Reply:
I think that `gh pr view <PR-url> --json reviewDecision` gives the right answer, no?

Reply:
I don't have experience with the gh cli. But we don't have (yet?) setup required reviews as a branch protection rule, so according to https://stackoverflow.com/a/74084046 `reviewDecision` is not set. The same answer describes also a solution on how to get all reviews and their state, so maybe use this?

Reply:
`reviewDecision` is probably also only taking reviews by people with write permission into account, at least the "Approved" flag does so as observed in https://github.com/sagemath/sage/pull/34983#issuecomment-1426655658.

Reply:
> I don't have experience with the gh cli. But we don't have (yet?) setup required reviews as a branch protection rule, so according to https://stackoverflow.com/a/74084046 `reviewDecision` is not set. The same answer describes also a solution on how to get all reviews and their state, so maybe use this?



> `reviewDecision` is probably also only taking reviews by people with write permission into account, at least the "Approved" flag does so as observed in [sagemath/sage#34983 (comment)](https://github.com/sagemath/sage/pull/34983#issuecomment-1426655658).

Thank you for these useful hints. I've only looked at examples where `reviewDecision` was specified (#35005 and #35022).

Reply:
I first check `reviewDecision`. If it is `None` I check if there isn't any review requesting changes and at least one approved review of the actor.